agents/chopper_mc2000b/chopper_mc2000b_agent.py

```diff
@@ -0,0 +1,339 @@
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            MC2000BSetFrequency(self.hdl, params['freq'])
+
+        return True, "Chopper frequency set to {} Hz".format(params['freq'])
+
+    @ocs_agent.param('bladetype', type=str, default='MC1F2')
+    def set_bladetype(self, session, params):
+        """set_bladetype(bladetype=None)
+
+        **Task** - Set the bladetype of the chopper. Selecting a bladetype
+        influences the range of frequencies permitted for the chopper.
+
+        Parameters
+        __________
+        bladetype : str
+            Name of bladetype assigned to chopper controller setup.
+            Default set to 'MC1F2' to reach the range of 4-8Hz.
+        """
+        with self.lock.acquire_timeout(timeout=3, job='set_bladetype') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            bladetype = bladetype_keys[params['bladetype']]
+            MC2000BSetBladeType(self.hdl, bladetype)
+
+        return True, "Chopper bladetype set to {}".format(params['bladetype'])
+
+    @ocs_agent.param('output_mode', type=str, choices=['actual', 'target'], default='target')
+    def set_reference_output_mode(self, session, params):
+        """set_reference_output_mode(output_mode=None)
+
+        **Task** - Set the output reference mode to determine the setting of
+        frequency output/input.
+
+        Parameters
+        __________
+        output_mode : str
+            Output reference mode of chopper frequency. Possible modes
+            are 'target' or 'actual'. Default set to 'target'.
+
+        """
+        with self.lock.acquire_timeout(timeout=3, job='set_reference_output_mode') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            mode = params['output_mode']
+            mode_int = outputmode_keys[mode]
+            MC2000BSetReferenceOutput(self.hdl, mode_int)
+
+        return True, "Chopper output mode set to {}".format(params['output_mode'])
+
+    @ocs_agent.param('reference', type=str, default='internalinner')
+    def set_blade_reference(self, session, params):
+        """set_blade_reference(reference=None)
+
+        **Task** - Set the reference mode for the blade. This is the point on
+        the chopper blades for the controller to measure and set frequency.
+
+        Parameters
+        __________
+        reference : str
```

Comment:
Similar to blade type, are there Thorlabs docs that describe valid reference modes? If not, we should list here.